src/server/cmd/pachd/main.go

```diff
@@ -102,18 +102,7 @@ func main() {
 	}
 }
 
-func doReadinessCheck(config interface{}) error {
-	env := serviceenv.InitPachOnlyEnv(serviceenv.NewConfiguration(config))
-	return env.GetPachClient(context.Background()).Health()
-}
-
-func doEnterpriseMode(config interface{}) (retErr error) {
-	defer func() {
-		if retErr != nil {
-			log.WithError(retErr).Print("failed to start server")
-			pprof.Lookup("goroutine").WriteTo(os.Stderr, 2)
-		}
-	}()
+func commonSetup(config interface{}, service string, initDB, initDexDB bool) (env *serviceenv.NonblockingServiceEnv, err error) {
```

Comment:
definitely a nit but I strongly dislike multiple boolean arguments. Here I guess it's simple as they are always the same value. From a readability perspective I'd prefer to pass them in a way that includes names, either through functional options or an options struct. Alternatively, maybe just make a separate "setUpDB" function that does both, since it's not strictly "common"